confmap/provider.go

```diff
@@ -143,6 +143,7 @@ func withStringRepresentation(stringRepresentation string) RetrievedOption {
 func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (*Retrieved, error) {
 	var rawConf any
 	if err := yaml.Unmarshal(yamlBytes, &rawConf); err != nil {
+		fmt.Printf("unmarshaling from yaml failed: %v\nAttempting to use it as a string verbatim. You can ignore this warning if the collector starts successfully.\n\n", err)
```

Comment:
I shouldn't use `fmt.Printf`. We need to print everything with configured loggers.

Reply:
Hmm. I was confused about the same.
I followed the same step as `config.Validate(..)`.

It prints a message if `service::telemetry::metrics` validation fails.

Needs some additional work to involve loggers in this section but I get your point.